sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Azure.Monitor.OpenTelemetry.Exporter.Tests/TelemetryItemTests.cs

```diff
@@ -427,100 +433,92 @@ public void OTelResourceMetricTelemetryHasAllResourceAttributes()
                 {
                     { "foo", "bar" }
                 };
 
-                using ActivitySource activitySource = new ActivitySource(ActivitySourceName);
-                using var activity1 = activitySource.StartActivity(
-                    ActivityName,
-                    ActivityKind.Server,
-                    null,
-                    startTime: DateTime.UtcNow);
+            using ActivitySource activitySource = new ActivitySource(ActivitySourceName);
+            using var activity1 = activitySource.StartActivity(
+                ActivityName,
+                ActivityKind.Server,
+                null,
+                startTime: DateTime.UtcNow);
+
+            Assert.NotNull(activity1);
 
-                Assert.NotNull(activity1);
+            var resource = ResourceBuilder.CreateEmpty().AddAttributes(testAttributes).Build();
+            var azMonResource = resource.CreateAzureMonitorResource(platform: GetMockPlatform("true"), instrumentationKey: instrumentationKey);
 
-                var resource = ResourceBuilder.CreateEmpty().AddAttributes(testAttributes).Build();
-                var azMonResource = resource.CreateAzureMonitorResource(instrumentationKey);
+            var telemetryItems = TraceHelper.OtelToAzureMonitorTrace(new Batch<Activity>(new Activity[] { activity1 }, 1), azMonResource, instrumentationKey, 1.0f);
+            var telemetryItem1 = telemetryItems.First(x => x.Name == "Metric"); // Collect the "_OTELRESOURCE_" metric.
 
-                var telemetryItems = TraceHelper.OtelToAzureMonitorTrace(new Batch<Activity>(new Activity[] { activity1 }, 1), null, instrumentationKey, 1.0f);
-                var telemetryItem1 = telemetryItems.FirstOrDefault();
+            var monitorBase = telemetryItem1?.Data;
+            var metricsData = monitorBase?.BaseData as MetricsData;
 
-                var monitorBase = telemetryItem1?.Data;
-                var metricsData = monitorBase?.BaseData as MetricsData;
+            Assert.NotNull(metricsData?.Metrics);
 
-                Assert.NotNull(metricsData?.Metrics);
+            var metricDataPoint = metricsData?.Metrics[0];
+            Assert.Equal("_OTELRESOURCE_", metricDataPoint?.Name);
+            Assert.Equal(1, metricsData?.Properties.Count);
+            Assert.Equal("bar", metricsData?.Properties["foo"]);
 
-                var metricDataPoint = metricsData?.Metrics[0];
-                Assert.Equal("_OTELRESOURCE_", metricDataPoint?.Name);
-                Assert.Equal(1, metricsData?.Properties.Count);
-                Assert.Equal("bar", metricsData?.Properties["foo"]);
+            // This test will simulate exporting a second batch of telemetry items.
+            // The timestamp on the second batch MUST NOT match the timestamp on the first batch.
+            // Here we force a 1 millisecond delay and evaulate that the timestamps are unique.
+            Task.Delay(1).Wait();
```

Comment:
This is a fix for a test bug. Ping me offline, we can debug and see this test fail in the `main` branch.